sdk/dotnet/Utils.cs

```diff
@@ -0,0 +1,133 @@
+        }
+
+        /// <summary>
+        /// Convert an argument array to an argument string for using with Process.StartInfo.Arguments.
+        /// </summary>
+        private static string EscapeArguments(string argument)
+        {
+            using var characterEnumerator = argument.GetEnumerator();
+            var escapedArgument = new StringBuilder();
+            var backslashCount = 0;
+            var needsQuotes = false;
+
+            while (characterEnumerator.MoveNext())
+            {
+                switch (characterEnumerator.Current)
+                {
+                    case '\\':
+                        // Backslashes are simply passed through, except when they need
+                        // to be escaped when followed by a \", e.g. the argument string
+                        // \", which would be encoded to \\\"
+                        backslashCount++;
+                        escapedArgument.Append('\\');
+                        break;
+
+                    case '\"':
+                        // Escape any preceding backslashes
+                        for (var c = 0; c < backslashCount; c++)
+                        {
+                            escapedArgument.Append('\\');
+                        }
+
+                        // Append an escaped double quote.
+                        escapedArgument.Append("\\\"");
+
+                        // Reset the backslash counter.
+                        backslashCount = 0;
+                        break;
+
+                    case ' ':
+                    case '\t':
+                        // White spaces are escaped by surrounding the entire string with
+                        // double quotes, which should be done at the end to prevent 
+                        // multiple wrappings.
+                        needsQuotes = true;
+
+                        // Append the whitespace
+                        escapedArgument.Append(characterEnumerator.Current);
+
+                        // Reset the backslash counter.
+                        backslashCount = 0;
+                        break;
+
+                    default:
+                        // Reset the backslash counter.
+                        backslashCount = 0;
+
+                        // Append the current character
+                        escapedArgument.Append(characterEnumerator.Current);
+                        break;
+                }
+            }
+
+            // No need to wrap in quotes
+            if (!needsQuotes)
+            {
+                return escapedArgument.ToString();
+            }
+
+            // Prepend the "
+            escapedArgument.Insert(0, '"');
+
+            // Escape any preceding backslashes before appending the "
+            for (var c = 0; c < backslashCount; c++)
+            {
+                escapedArgument.Append('\\');
+            }
+
+            // Append the final "
+            escapedArgument.Append('\"');
```

Comment:
nit: this duplicates your switch case above.  consider a helper.

Reply:
What do you mean here?

Reply:
basically, ths looks duplicated with the code here: https://github.com/pulumi/pulumi-docker/pull/115/files/45dae59a6d7429ae40a35a0cff89ca290d5b5fc1#diff-6a4e8545028a22013e41ffad57a63628R76

Reply:
Gotcha!

Reply:
These lines are a bit different though. I refactored the loops to one-liners.